doc/users_guide.rst

```diff
@@ -0,0 +1,306 @@
+  The closest thing to it is ``"name"`` which is just a pretty-printing name and
+  has no extra impact.
+* You don't have to specify devlib modules to load anymore. All (loadable)
+  modules are now loaded. If you find some module too slow to load, you can
+  specify a list of modules to exclude.
+* LISA used to have ``target.config`` JSON file at its root. Its equivalent is
+  now ``target_conf.yml``, which is in YAML.
+
+.. admonition:: Cool new feature
+
+   :class:`lisa.env.TestEnv` instances can now be easily be  created
+   :meth:`via CLI<lisa.env.TestEnv.from_cli>`!
+
+
+Trace
++++++
+
+The :class:`lisa.trace.Trace` class hasn't changed much in terms of functionality,
+but we did rename/move things to make them more coherent.
+
+* Removed last occurences of camelCase
+* Removed big.LITTLE assumptions and made the code only rely on CPU capacities or
+  frequency domains, where relevant.
+* ``Trace.data_frame`` is gone:
+
+**LISA legacy**::
+
+  trace.data_frame.trace_event("sched_switch")
+
+**LISA next**::
+
+  trace.df_events("sched_switch")
+
+
+Analysis
+++++++++
+
+Analysis modules must now be called explicitely:
+
+**LISA legacy**::
+
+  trace.data_frame.latency_df("test")
+
+**LISA next**::
+
+  trace.analysis.latency.df_latency("test")
+
+.. admonition:: Cool new feature
+
+  Trace events required by the analysis methods are now automatically documented,
+  see :meth:`~lisa.analysis.frequency.FrequencyAnalysis.df_cpu_frequency_residency`
+  for instance.
+
+wlgen
++++++
+
+The :class:`lisa.wlgen.rta.RTA` class has been simplified somewhat:
+
+* :class:`lisa.wlgen.rta.RTATask` no longer has a superfluous ``get()`` method
+* ``RTA.conf()`` has been squashed inside alternative constructors, see
+  :meth:`lisa.wlgen.rta.RTA.by_str` and :meth:`lisa.wlgen.rta.RTA.by_profile`.
+
+**LISA legacy**::
+
+  profile = {}
+  profile["my_task"] = Periodic(duty_cycle_pct=30).get()
+
+  wload = RTA(te.target, "foo", calibration)
+  wload.conf(kind='profile', params=profile)
+
+**LISA next**::
+
+  profile = {}
+  profile["my_task"] = Periodic(duty_cycle_pct=30)
+
+  wload = RTA.by_profile(te, "foo", profile, res_dir, calibration)
+
+Kernel tests
+++++++++++++
+
+The ``Executor`` from LISA legacy has been entirely removed, and a new test
+framework has been put in place. Tests are now coded as straight Python classes,
+which means they can be imported and executed in scripts/notebooks without any
+additionnal effort. See :ref:`kernel-testing-page`.
+
+Notebooks
++++++++++
+
+In LISA legacy, notebooks served as documentations and where the main source of
+exemples. We now have a proper documentation (you're reading it!), so we
+greatly trimmed down the number of notebooks we had.
+
+We've kept older notebooks in ``ipynb/deprecated``, but they have not been ported
+over to the new APIs (or even to Python3) so they won't work. They are there in
+case we find a reason to bring back some of them.
```

Comment:
Thinking about it, this is a bit odd actually no ? Maybe we should just remove them ? Keeping things that are totally borked feels kinda wrong. I mean, they will still exist in the 'old' Lisa I suppose, and we can always go back there to pick them up if we want to ... Thoughts ?

Reply:
Mmm so the main reason I kept them is to not lose their history. That way we still have the right contributors associated to the right notebooks - I tried to keep that consistent with as many files as possible, no matter how much we moved them. Deleting them, then reverting the delete + changing them would be a bit of a pain (plus I wonder if the revert wouldn't cover the whole file when blaming it :/)